File: java/src/test/java/problems101_200/MaximumProductSubarrayTest.java
package problems101_200;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class MaximumProductSubarrayTest extends SampledTest {
    @Test
    public void test() {
        for (MaximumProductSubarray solution : MaximumProductSubarray.values()) {
            for (MaximumProductSubarraySample sample
                : getSamples(MaximumProductSubarraySample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.maxProduct(sample.input.nums))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems101_200/MaximumProductSubarrayTest.kt
package problems101_200

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import kotlin.test.Test

class MaximumProductSubarrayTest : SampledTest() {
    @Test
    fun test() =
        MaximumProductSubarray.entries.forEach { solution ->
            getSamples(Array<MaximumProductSubarraySample>::class.java)
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(solution.maxProduct(input.nums))
                        .isEqualTo(output)
                }
        }
}
